### PR TITLE
[Invalid PR] Added missing support to listen to step events within workers

### PR DESCRIPTION
**Pull request by abhimanyupandian**
## Motivation/Description of the PR
Currently, only events for suite and test can be listened to when tests are run in parallel using workers.
This PR fixes the missing implementations for step events.

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [ ] Playwright

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] stepTimeout
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [x] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by abhimanyupandian** [comment]
@DavertMik could you please review and merge this bug fix?
The UTs are failing due to someother reason and not due to the changes in this fix.

File: lib/command/workers/runTests.js [hunk 1]
```diff
@@ -134,6 +167,12 @@ function initializeListeners() {
   // calculate duration
   event.dispatcher.on(event.test.started, test => test.start = new Date());
 
+  // tests
```
Comment (DavertMik) [inline lib/command/workers/runTests.js]: steps

**Review by DavertMik — COMMENTED** [review]
Sorry, but what problem does this solve?

If the purpose is to send these events to print out steps - it won't work. Because if we will print all steps from all workers it will produce a total mess, so you won't be able to read the scenario, anyway/

For this case I don't see a way it can be implemented.

Maybe there are cases I'm currently missing?

**Comment by abhimanyupandian** [comment]
> Sorry, but what problem does this solve?
> 
> If the purpose is to send these events to print out steps - it won't work. Because if we will print all steps from all workers it will produce a total mess, so you won't be able to read the scenario, anyway/
> 
> For this case I don't see a way it can be implemented.
> 
> Maybe there are cases I'm currently missing?

@DavertMik 

Although it is anyways not possible to print them out as we have the stdout disabled for parallel tests, the intent is not to print the steps from all workers. 

Currently, when we run tests in parallel, we are unable to capture the step events (finished/started/passed/failed). However, the events for suite and test work just fine. 

Consider the following case with the 'commentStep' plugin enabled:
```typescript
const changeABC = () => step`Save after changing Profile to ABC`;
function save() {
      __changeABC;
      I.selectOption('enable');
      I.click('Save');
      I.waitForText('Saved successfully');
}
```

**We have a plugin that keeps track of the steps that have failed** (something like {stepTitle: count}).  For non-parallel execution, this works fine. It will be useful if we can get the details of the step (here the title 'Apply Changes after Market Change' is useful) for parallel tests also, within a custom plugin just like we have for non-parallel tests.

That is what this PR intends to do.

It would be expected/nice to have the same behavior for parallel tests as we are unable to run tests in parallel due to this.

**Comment by abhimanyupandian** [comment]
Hey @DavertMik, could you please help us with this PR. We are unable to fully support parallel tests due to the missing event listeners for steps in parallel tests

**Comment by abhimanyupandian** [comment]
@DavertMik screwed something up as part of the commits :(
Please confirm if this PR can be deleted, will raise a support ticket for the same.

**Pull request closed without merging**